### DISCUSSION TITLE
Current Weather Not Accurate

### DISCUSSION THREAD

**Discussion by cloneofghosts** [Pirate-Weather/pirateweather#222] [General]
I've noticed today that the weather data from PW seems to be more off than it has been the last few days. The cloud cover has been pretty off since about 12pm (the cloud cover in HRRR is better than in NBM currently) and while the temperature is better now it's also been off since around the same time. There's also been two occasions that the API has returned rain when it didn't rain.

PW:
![image](https://github.com/Pirate-Weather/pirateweather/assets/10248058/bf8bb50c-5570-4831-9b0c-9807f3b672a4)

EC:
![image](https://github.com/Pirate-Weather/pirateweather/assets/10248058/9220d59d-89d9-4825-8c63-0a0cd3621037)

I know there's not much you can do without integrating more models but I'd thought to post this anyway.

**Comment by alexander0042** [comment]
Yea, I noticed today as well! Weird weather day today, with a hot and sunny morning followed by a cold front, but now looking nicer again? 

- One thing I like to do when this sort of weirdness happens is check out SpotWX, which shows a few models I don't have integrated yet: [RDPS](https://spotwx.com/products/grib_index.php?model=hrdps_continental&lat=45.45202&lon=-75.62214&tz=America/Toronto&label=)
- [HRRR](https://spotwx.com/products/grib_index.php?model=hrrr_wrfprsf&lat=45.45202&lon=-75.62214&tz=America/Toronto&label=)

It's only real-time, so I have to catch it in the moment, but it provide a nice control point. Moreover, PW's `exclude=nbm` should match the HRRR results from SpotWX, although sometimes off by a degree due to interpolations

**Reply by cloneofghosts** [reply]
Thanks for the link. I checked out the HRRR data and it seems to be pretty similar to the data shown in PW so it's not a processing issue at least. I wonder if the cloud cover issue in NBM is because the cloud cover wasn't expected and while the rapidly updating models show cloud cover the ones which update more slowly aren't. Whatever magic is used to blend them all together it's pulling in the cloud cover more from the ones that are showing no clouds rather than the rapid updating ones?

**Reply by cloneofghosts** [reply]
Am seeing this again this afternoon with cloud cover as temperature seems to be fine today. HRRR and NBM are giving similar results to eachother and the API is giving similar results to what SpotWX shows for my location. If I take a look at the different models available on that website these ones are giving more accurate results:

- HRDPS
- RAP
- GFS

**Reply by cloneofghosts** [reply]
Seems like it fixed itself over the weekend so will close.

**Comment by cloneofghosts** [comment]
@alexander0042 Noticed the last few days that the daytime temperature has been much lower compared to other sites.

PW:
![image](https://github.com/user-attachments/assets/786e8aca-6ad4-400f-a32d-5cd8032c91b2)

EC:
![image](https://github.com/user-attachments/assets/6dbcc71e-c193-4fbe-8d1b-26dc9c7e835d)

If I change my query to use the same co-ordinates as EC I only get marginally higher temperatures. I looked at SpotWX for my location and it shows this:

![image](https://github.com/user-attachments/assets/4732e431-d0e3-495d-9c4e-b203b01586ba)![image](https://github.com/user-attachments/assets/19de22c3-83ad-4401-a7ae-e582522c1123)

While the hourly data in PW isn't exactly the same as SpotWX but it's close enough to assume the issue isn't with PW but probably due to the 2h lag time that HRRR has.

**Reply by alexander0042** [reply]
Interesting! You're talking about at the "currently" temperatures right, not the daily max/ min ones, since those are looking pretty close to me at the moment. 

SpotWX shows the model date up at the top of the page, so you should be able to compare if it's the same run as PW is looking at. sub-hourly HRRR isn't available on there; however, it's the same model run as HRRR, so really should be comparable at the hourly steps. 

It would be interesting to look at the difference between HRRR and NBM for temperature. If NBM is consistently more accurate, it might make sense to use it as the source for currently temperatures, since they really don't benefit as much from the increased temporal resolution of sub-hourly

**Reply by cloneofghosts** [reply]
Yes, I was looking at the current temperatures and not the daily high/low temperatures. It's much closer now to the temperatures at the airport than it was this morning so maybe the temperatures were rising too fast for HRRR to keep up?

The run I was looking at in SpotWX was the 9am HRRR run which should be the same run in PW at the time (or maybe PW was 1h behind because of the high/low changes?).

It would be interesting to compare HRRR/NBM to the temperatures reported at the airport to see which model gets closer. Though the upside to using the sub-hourly HRRR model is that its integrated into the API the quickest so it can react faster to sudden storms/precipitation.

**Reply by cloneofghosts** [reply]
@alexander0042 Seems like whatever the issue was got fixed on Sunday as temperatures are much closer to the airport temperatures (even slightly above at times).
